src/components/MenuEnvelope.vue

```diff
@@ -567,8 +568,45 @@ export default {
 		setCustomSnooze() {
 			this.onSnooze(this.customSnoozeDateTime.valueOf())
 		},
-		onPrint() {
-			window.print()
+		async onPrint() {
+			await this.$emit('print')
+
+			const printStyles = `
+				* {
+					font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Cantarell, Ubuntu, 'Helvetica Neue', Arial, 'Noto Color Emoji', sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol';
+				};
+			`
+
+			let combinedContent = ''
+			const messageFrames = document.querySelectorAll('iframe.message-frame')
+
+			messageFrames.forEach(frame => {
+				const doc = frame.contentDocument || frame.contentWindow.document
+				combinedContent += doc.body.innerHTML
+			})
+
+			const frame = document.createElement('iframe')
+			frame.name = 'printFrame'
+			frame.style.position = 'absolute'
+			frame.style.top = '-1000000px'
+			document.body.appendChild(frame)
+			const frameDoc = frame.contentWindow ? frame.contentWindow : frame.contentDocument.document ? frame.contentDocument.document : frame.contentDocument
+			frameDoc.document.open()
+			frameDoc.document.write('<html><head><style>')
+			frameDoc.document.write(printStyles)
+			frameDoc.document.write('</style></head><body>')
+			frameDoc.document.write(document.querySelector('#mail-thread-header h2').outerHTML)
+			frameDoc.document.write(`<h4>${this.envelope.from[0].label}, ${this.envelope.from[0].email}</h4>`)
```

Comment:
you have to sanitize user input before putting it into DOM. the label could contain HTML markup

Reply:
I added a method to do that, but tell me if we have a more standard way somewhere